examples/aflow/scripts/evaluator.py

```diff
@@ -0,0 +1,176 @@
+            va_list = None
+        else:
+            data_path = "examples/aflow/data/human-eval_validate.jsonl"  # Replace with your JSONL file path
+            va_list = [1,2,3] # Replace with the filtered index list
+
+        graph = await load_graph()
+        
+        avg_score, avg_cost, total_cost = await optimize_humaneval_evaluation(graph, data_path, path, va_list)
+        
+        return avg_score, avg_cost, total_cost
+
+    async def _mbpp_eval(self, graph_class, params, path, test=False):
+        """
+        Evaluate MBPP dataset.
+        """
+        async def load_graph():
+            dataset = params["dataset"]
+            llm_config = params["llm_config"]
+            return graph_class(name="MBPP", llm_config=llm_config, dataset=dataset)
+        
+        if test:
+            data_path = "examples/aflow/data/mbpp_test.jsonl"
+            va_list = None 
+        else:
+            data_path = "examples/aflow/data/mbpp_validate.jsonl"
+            va_list = [1,2,3] # Replace with the filtered index list
+
+        graph = await load_graph()
+        
+        avg_score, avg_cost, total_cost = await optimize_mbpp_evaluation(graph, data_path, path, va_list)
+        
+        return avg_score, avg_cost, total_cost
+
+    async def _hotpotqa_eval(self, graph_class, params, path, test=False):
+        """
+        Evaluate HotpotQA dataset.
+        """
+        async def load_graph():
+            dataset = params["dataset"]
+            llm_config = params["llm_config"]
+            return graph_class(name="HotpotQA", llm_config=llm_config, dataset=dataset)
+        
+        if test:
+            data_path = "examples/aflow/data/hotpotqa_test.jsonl"
+            va_list = None
+        else:
+            data_path = "examples/aflow/data/hotpotqa_validate.jsonl"
+            va_list = [1,2,3] # Replace with the filtered index list
+
+        graph = await load_graph()
+        
+        avg_score, avg_cost, total_cost = await optimize_hotpotqa_evaluation(graph, data_path, path, va_list)
+        
+        return avg_score, avg_cost, total_cost
+    
+
+    async def _drop_eval(self, graph_class, params, path, test=False):
+        """
+        Evaluate DROP dataset.
+        """
+        async def load_graph():
+            dataset = params["dataset"]
+            llm_config = params["llm_config"]
+            return graph_class(name="DROP", llm_config=llm_config, dataset=dataset)
+        
+        if test:
+            data_path = "examples/aflow/data/drop_test.jsonl"
+            va_list = None
+        else:
+            data_path = "examples/aflow/data/drop_validate.jsonl"
+            va_list = [1,2,3] # Replace with the filtered index list
+
+        graph = await load_graph()
+        
+        avg_score, avg_cost, total_cost = await optimize_drop_evaluation(graph, data_path, path, va_list)
+        
+        return avg_score, avg_cost, total_cost
```

Comment:
Code duplication is too high, it is recommended to refactor

Reply:
have refactored